src/registry/routes/components.js

```diff
@@ -68,8 +68,13 @@ module.exports = function(conf, repository) {
         );
       },
       (err, results) => {
-        setHeaders(results, res);
-        res.status(200).json(results);
+        try {
+          setHeaders(results, res);
+          res.status(200).json(results);
+        } catch (e) {
+          console.log(results);
+          res.status(500).json({});
```

Comment:
Can we return the error in the form of `returnError` for consistency?
```
res.json({
  code, // Sort of indexable thing like render_error
  error // A more descriptive thing perhaps with the component name and the version?
})

Reply:
I could do the following to give better insight server-side & client side: 

```try {
          setHeaders(results, res);
          res.status(200).json(results);
        } catch (e) {
          console.log('RENDER_ERROR', format(
            strings.errors.registry.RENDER_ERROR,
            results.name,
            results.requestVersion),
            results.code, results.error);

          if (results.code && results.error) {
            res.status(500).json({"code": results.code, "error": results.error });
          } else {
            res.status(500).json({"code": "RENDER_ERROR", "error": format(
              strings.errors.registry.RENDER_ERROR,
                results.name,
                results.requestVersion
              )});
          }
        }
```

The is the current results payload that we are seeing in our logs when this exception is caught:
```
{ 
       code: 'GENERIC_ERROR',
       error: 'Component execution error: Request failed with status code 404',
       details: [Object],
       name: 'poorly-written-component',  //name changed to protect the innocent 
       requestVersion: '1.3.42' }
```

Reply:
I think that would definitely improve, together with a test